File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const spg1 = [1, 2, 3, 4, 5];
    const spg2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const spg3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    const cnt1 = answers.filter((x, i) => x===spg1[i%spg1.length]).length;
    const cnt2 = answers.filter((x, i) => x===spg2[i%spg2.length]).length;
    const cnt3 = answers.filter((x, i) => x===spg3[i%spg3.length]).length;
    
    const maxScore = Math.max(cnt1, cnt2, cnt3);
    
    let result = [];
    
    if(maxScore<=cnt1) result.push(1);
    if(maxScore<=cnt2) result.push(2);
    if(maxScore<=cnt3) result.push(3);
    
    return result;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    s1, s2, s3 = [1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    cnt = [0, 0, 0]
    answer = []

    for i, a in enumerate(answers):
        if a == s1[i % len(s1)]:
            cnt[0] += 1
        if a == s2[i % len(s2)]:
            cnt[1] += 1
        if a == s3[i % len(s3)]:
            cnt[2] += 1

    max_score = max(cnt)

    for i, c in enumerate(cnt):
        if c >= max_score:
            answer.append(i + 1)
            
    return answer